proposals/artifacts.md

```diff
@@ -0,0 +1,58 @@
+# OCI artifact project proposal
+
+## Abstract
+Container registries, implementing the [distribution-spec][distribution-spec], provide reliable, highly scalable, secured storage services for container images. Customers either use a cloud provider implementation, vendor implementations, or instance the open source implementation of distribution. They configure security and networking to assure the images in the registry are locked down and accessible by the resources required. Cloud providers and vendors often provide additional values atop their registry implementations from security to productivity features. 
+
+Applications and services typically require additional artifacts to deploy and manage, including [helm](https://helm.sh) for deployment and [Open Policy Agent (OPA)](https://github.com/open-policy-agent/opa/issues/1413) for policy enforcement.
+
+Utilizing the [manifest][image-manifest] and [index][image-index] definitions, new artifacts can be stored and served using the [distribution-spec][distribution-spec] without changing the actual distribution spec. This repository will provide  a reference for artifact authors and registry implementors for supporting new artifact types with the existing implementations of distribution.
+
+By providing an OCI artifact definition, the community can continue to innovate, focusing on new artifact types without having to build yet another storage solution (YASS). 
+
+## Proposal
+Under the http://github.com/opencontainers organization:
+
+- Create a new **artifacts** repository, named http://github.com/opencontainers/artifacts
+- Update [distribution][distribution-spec] to generically reference [manifest][image-manifest] and [index][image-index], with image as one of many artifact types it can store. 
```

Comment:
The only thing we should be careful about (and this doesn't need to be included in this proposal -- it's just something to keep in mind when the distribution-spec changes are sent) is that we don't want to end up requiring all tools which pull images from distribution-spec to support all possible artifact types.

We don't have an existing concept of an IANA-like registry in OCI, but this new repo will effectively act as one (MIME type definitions without any requirements that implementations must or even should implement them).

Reply:
Yup, agree. The goal of the artifact definitions area will be to state what `mediaType`s the tooling around that artifact MUST, SHOULD or MAY support. For instance the docker cli SHOULD NOT need to know about Helm. The spec already states it should ignore types it doesn't know about.
What's interesting is how a new breed of generic registry tooling *could* benefit from these mediaTypes definitions.